src/_guides/language/effective-dart/design.md

```diff
@@ -1005,90 +1089,254 @@ var names = people.map((Person person) {
+    not its *type*. Given:
+
+    {:.bad-style}
+    <?code-excerpt "misc/lib/effective_dart/design_bad.dart (typedef-param)"?>
+    {% prettify dart %}
+    typedef bool TestNumber(num);
+    {% endprettify %}
+
+    Most users expect this to be a function type that takes a `num` and returns
+    `bool`. It is actually a function type that takes *any* object (`dynamic`)
+    and returns `bool`. The parameter's *name* (which isn't used for anything
+    except documentation in the typedef) is "num". This has been a
+    long-standing source of errors in Dart.
+
+The new syntax looks like:
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (new-typedef)"?>
+{% prettify dart %}
+typedef Comparison = int Function<T>(T, T);
+{% endprettify %}
+
+The new syntax can express anything the old syntax could express and more, and
+lacks the error-prone misfeature where a single identifier is treated as the
+parameter's name. The same function type syntax after the `=` in the typedef is
+also allowed anywhere a type annotation may appear, giving us a single
+consistent way to write function types anywhere in a program.
+
+The old typedef syntax is still supported to avoid breaking existing code, but
+it's deprecated.
+
+
+### PREFER inline function types over typedefs.
+
+In Dart 1, if you wanted to use a function type for a field, variable, or
+generic type argument, you had to first define a typedef for it. Dart 2 supports
+a function type syntax that can be used anywhere a type annotation is allowed:
+
+{:.good-style}
+<?code-excerpt "misc/lib/effective_dart/design_good.dart (function-type)"  replace="/(bool|void) Function\(Event\)/[!$&!]/g"?>
+{% prettify dart %}
+class FilteredObservable {
+  final [!bool Function(Event)!] _predicate;
+  final List<[!void Function(Event)!]> _observers;
+
+  FilteredObservable(this._predicate, this._observers);
+
+  [!void Function(Event)!] notify(Event event) {
+    if (!_predicate(event)) return null;
+
+    [!void Function(Event)!] last;
+    for (var observer in _observers) {
+      observer(event);
+      last = observer;
+    }
+
+    return last;
+  }
+}
+{% endprettify %}
+
+It may still be worth defining a typedef if the function type is particularly
+long or frequently used. But in most cases, users want to see what the function
+type actually is right where it's used, and the function type syntax gives them
+that clarity.
+
+
+### CONSIDER using the generalized function type syntax for parameters.
+
+Dart has a special syntax when defining a parameter whose type is a function.
+Sort of like C, you surround the parameter's name with the function's return
+type and parameter signature:
+
+<?code-excerpt "misc/lib/effective_dart/design_bad.dart (function-type-param)"?>
+{% prettify dart %}
+Iterable<T> where(bool predicate(T element)) => ...
+{% endprettify %}
+
+Before Dart 2 added general function type syntax, this was the only way to give
```

Comment:
general -> the general
?
(I stopped and reread due to the "the" being gone. Maybe you meant that, though.)

Reply:
Done. I really wish there was a canonical name for this syntax. I made up "generalized function type syntax", but that's a mouthful. :-/